### PR TITLE
[ci][byod/8] control the list of pip through source control

### PR DESCRIPTION
**Pull request by can-anyscale**
## Why are these changes needed?
**Context**: This is a series of PR that will allow OSS release tests to run on anyscale byod images. Currently, it runs against nightly anyscale which can be unreliable. Using byod image, we will be able to run them against staging or production anyscale. Proposal doc: https://docs.google.com/document/d/12WU4vpR0jHgOwceoSgPtE4B-B-6mBonnHHiCSMmMNoY/edit#

- Copy-cat the trick @aslonnie has to control the list of pip dependencies through source control. This way, test reliability is resistant to external library updates. Any update needs to perform through a code change/source control.
- Use another byod test for testing. The existing gpu test use ray[tune] which dependencies I don't want to deal with right now.

## Checks

- [X] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [X] I've run `scripts/format.sh` to lint the changes in this PR.
- Testing Strategy
   - [X] Unit tests
   - [X] Release tests - https://buildkite.com/ray-project/release-tests-pr/builds/39315#_

### CODE REVIEW

File: release/ray_release/byod/requirements_byod.txt [hunk 1]
```diff
@@ -1,7 +1,508 @@
...
+    --hash=sha256:b02a7ffeaa61448737f39a4210b8ee60234bda0515a0c0d8562f884454105b0f \
+    --hash=sha256:b85f98d4cc2f72452f6792ab4463a3541bc5678a8cdd3da0e139ba2fe8b56d42 \
+    --hash=sha256:c07071bc8d02aa8fcdfe139ab6a1ef57d3b64c9e30e84d12d45c9f4d89fb6536 \
+    --hash=sha256:c55f9889e436f14b4f84a9c00ebad0d31f5b4626f10cf8018e6c676f92a6d199
+    # via -r release/requirements_byod.in
+typing-extensions==4.5.0 \
+    --hash=sha256:5cb5f4a79139d699607b3ef622a1dedafa84e115ab0024e0d9c044a9479ca7cb \
+    --hash=sha256:fb33085c39dd998ac16d1431ebc293a8b3eedd00fd4a32de0ff79002c19511b4
+    # via torch
+urllib3==1.26.15 \
```
Comment (can-anyscale) [inline release/ray_release/byod/requirements_byod.txt:503]: ah ha

File: release/release_tests.yaml [hunk 2]
```diff
@@ -4466,8 +4468,7 @@
   team: core
   frequency: manual
   working_dir: microbenchmark
-
-  python: "3.7"
+  python: "3.8"
```
Comment (aslonnie) [inline release/release_tests.yaml:4471]: why this change? is this intended?

File: release/release_tests.yaml [hunk 3]
```diff
@@ -1801,6 +1801,30 @@
 
   alert: tune_tests
 
+- name: tune_cloud_durable_upload_rllib_str_byod
+  group: Tune cloud tests
+  working_dir: tune_tests/cloud_tests
+  python: "3.8"
```
Comment (aslonnie) [inline release/release_tests.yaml:1807]: are we moving everything to py38 yet?

Comment (can-anyscale) [reply release/release_tests.yaml:1807]: yes, the list of pip cannot resolve with python 3.7 lol

**Review by aslonnie — APPROVED** [review]

**Comment by ollie-iterators** [comment]
I belive the documentation error can be fixed by merging in recent changes from the main branch

**Comment by aslonnie** [comment]
> I belive the documentation error can be fixed by merging in recent changes from the main branch

which doc error? could you be more specific? why do you think it can be fixed by merging in recent changes?

**Comment by ollie-iterators** [comment]
> > I belive the documentation error can be fixed by merging in recent changes from the main branch
> 
> which doc error? could you be more specific? why do you think it can be fixed by merging in recent changes?

Because the Documentation test in https://flakey-tests.ray.io/ is not failing right now.

**Comment by aslonnie** [comment]
> > > I belive the documentation error can be fixed by merging in recent changes from the main branch
> > 
> > 
> > which doc error? could you be more specific? why do you think it can be fixed by merging in recent changes?
> 
> Because the Documentation test in https://flakey-tests.ray.io/ is not failing right now.

thanks. could you add that related supporting evidence in the first comment next time? :) preferably with a link to the build, like this:

https://buildkite.com/ray-project/oss-ci-build-branch/builds/4127#01885492-beb6-44c1-9c58-2378387eefcd
